contracts/CellarPoolShare.sol

```diff
@@ -451,9 +464,32 @@ contract CellarPoolShare is ICellarPoolShare, BlockLock {
                 recipient: address(this),
                 deadline: type(uint256).max
             });
-        (uint256 outAmount0, uint256 outAmount1, uint128 liquiditySum, uint256 fee0, uint256 fee1) =
+
+        (uint256 outAmount0, uint256 outAmount1, uint128 liquiditySum, CellarFees memory cellarFees) =
             _removeLiquidity(removeParams);
         lastManageTimestamp = block.timestamp;
+
+        uint256 fee0 = cellarFees.management0 + cellarFees.performance0;
+        uint256 fee1 = cellarFees.management1 + cellarFees.performance1;
+        if (fee0 > cellarFees.collect0) {
+            fee0 = cellarFees.collect0;
+            if (cellarFees.management0 < cellarFees.collect0) {
+                cellarFees.performance0 = cellarFees.collect0 - cellarFees.management0;
+            } else {
+                cellarFees.management0 = cellarFees.collect0;
+                cellarFees.performance0 = 0;
+            }
+        }
+        if (fee1 > cellarFees.collect1) {
+            fee1 = cellarFees.collect1;
+            if (cellarFees.management1 < cellarFees.collect1) {
+                cellarFees.performance1 = cellarFees.collect1 - cellarFees.management1;
+            } else {
+                cellarFees.management1 = cellarFees.collect1;
+                cellarFees.performance1 = 0;
+            }
+        }
```

Comment:
This is great. But why this is only here, not in reinvest?
Or can you calculate on off-chain side? I am afraid that it increases gas and over the contract size.